src/passes/GlobalTypeOptimization.cpp

```diff
@@ -205,9 +204,9 @@ struct GlobalTypeOptimization : public Pass {
     SubTypes subTypes(*module);
     StructUtils::TypeHierarchyPropagator<FieldInfo> propagator(subTypes);
     auto dataFromSubsAndSupersMap = combinedSetGetInfos;
-    propagator.propagateToSuperAndSubTypes(dataFromSubsAndSupersMap);
+    propagator.propagateToSuperAndSubTypesWithExact(dataFromSubsAndSupersMap);
     auto dataFromSupersMap = std::move(combinedSetGetInfos);
-    propagator.propagateToSubTypes(dataFromSupersMap);
+    propagator.propagateToSubTypesWithExact(dataFromSupersMap);
```

Comment:
Same issue as before - why do we want exactness here? On the one hand it seems obvious we do, but it isn't the default - why not?

And, I see no test changes from this - if it doesn't help, why do it?

Reply:
Makes sense, but that sounds like it applies to all users of the utility? (why do it here and not in all others?)